NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -313,6 +387,105 @@ def read_CORSIKA7(input_file, declination=None, site=None):
 
     return evt
 
+def write_CORSIKA7(evt, output_file, declination=None, site=None):
+    """
+    Writes a NuRadioReco Event object to an HDF5 file in CORSIKA/CoREAS format,
+    reversing the operation of read_CORSIKA7().
+
+    Parameters
+    ----------
+    evt : NuRadioReco.framework.event.Event
+        The Event to store.
+    output_file : str
+        Path to the output HDF5 file.
+    declination : float, optional
+        Declination of the magnetic field in degrees. If None, determined from site.
+    site : str, optional
+        Name of the site, used to set default magnetic field and declination.
+    """
+    if os.path.exists(output_file):
+        raise FileExistsError(f"File '{output_file}' already exists. Aborting.")
+
+    if declination is None:
+        if site is not None:
+            try:
+                magnet = hp.get_magnetic_field_vector(site)
+                declination = hp.get_declination(magnet)
+                logger.info(
+                    f"Declination obtained from site information, is set to {declination / units.degree:.2f} deg"
+                )
+            except KeyError:
+                declination = 0
+                logger.warning(
+                    "Site is not recognised by radiotools. Defaulting to 0 degrees declination. "
+                    "This might lead to unexpected electric field polarizations."
+                )
+        else:
+            declination = 0
+            logger.warning(
+                "No declination or site given, assuming 0 degrees. "
+                "This might lead to unexpected electric field polarizations."
+            )
+        
+    with h5py.File(output_file, "x") as f:
+        
+        inputs_grp = f.create_group("inputs")
+        inputs_grp.attrs["RUNNR"] = evt.get_run_number()
+
+        sim_shower = evt.get_first_sim_shower()        
+        
+        inputs_grp.attrs["PRMPAR"] = sim_shower.get_parameter(shp.primary_particle)
+        inputs_grp.attrs["OBSLEV"] = sim_shower.get_parameter(shp.observation_level) / units.cm
+        energy_GeV = sim_shower.get_parameter(shp.energy) / units.GeV
+        inputs_grp.attrs["ERANGE"] = np.array([energy_GeV, energy_GeV])
+        inputs_grp.attrs["MAGNET"] = sim_shower.get_parameter(shp.magnetic_field_vector)
+
+        zenith = sim_shower.get_parameter(shp.zenith)
+        azimuth = sim_shower.get_parameter(shp.azimuth)
+        inputs_grp.attrs["PHIP"] = zenith
+        inputs_grp.attrs["THETAP"] = azimuth
```

Comment:
dont we have to account for the different coordinate system in the azimuth angle?